packages/snaps-utils/src/snaps.test.ts

```diff
@@ -35,7 +46,149 @@ describe('isCaipChainId', () => {
     'cosmos:iov-mainnet',
     'bip122:000000000019d6689c085ae165831e93',
     'cosmos:cosmoshub-2',
-  ])('returns true for valid IDs', (value) => {
+  ])('returns true for valid IDs (#%#)', (value) => {
     expect(isCaipChainId(value)).toBe(true);
   });
 });
+
+describe('LocalSnapIdStruct', () => {
+  it.each([
+    'local:http://localhost',
+    'local:http://localhost/',
+    'local:http://localhost/',
+    'local:https://localhost',
+    'local:https://localhost/',
+    'local:http://127.0.0.1/foo/bar',
+    'local:http://127.0.0.1:8080/',
+    'local:http://localhost:8080/',
+    'local:http://[::1]:8080/',
+    'local:http://[::1]',
+    'local:https://foo@localhost/',
+    'local:http://foo:bar@127.0.01/',
+  ])('validates "%s" as proper local ID', (value) => {
+    expect(is(value, LocalSnapIdStruct)).toBe(true);
+  });
+
+  it.each([
+    0,
+    1,
+    '',
+    'foo',
+    false,
+    true,
+    {},
+    [],
+    uri,
+    URL,
+    new URL('http://127.0.01'),
+    new URL('local:127.0.0.1'),
+    'http://localhost',
+    '127.0.0.1',
+    'local:127.0.0.1',
+    'local:127.0.0.1/',
+    'local:foo://127.0.0.1/',
+    'local:http://github.com',
+    'local:http://localhost/foo?bar=true',
+    'local:http://localhost/foo#bar',
+    'local:http://localhost/42?foo=true#bar',
+    'local',
+    'local:',
+    'local:http://',
+  ])('invalidates an improper local ID (#%#)', (value) => {
+    expect(is(value, LocalSnapIdStruct)).toBe(false);
+  });
+});
+
+describe('NpmSnapIdStruct', () => {
+  it.each([
+    'npm:foo',
+    'npm:foo-bar',
+    'npm://registry.com/foo',
+    'npm:@foo/bar',
+    'npm://registry.com/@foo/bar',
+    'npm://user@registry.com/bar',
+    'npm://user:pass@registry.com/bar',
+    'npm://[::1]/bar',
```

Comment:
We could, though it won't be used in this PR, so might as well leave it out for the future.

Reply:
I just meant to be on the safe side we could assert that our NPM ids are currently just `npm:VALID_NPM_ID`, since we don't need support for custom registries right now

Reply:
It's already secure - we have tests checking this in NpmLocation, and this would add more tests and implementation to an already big PR. I'm going to leave it out of this PR for now